functions/src/events/events.ts

```diff
@@ -0,0 +1,150 @@
+/**
+ * Events
+    Workshop 14
+        Title
+        Description
+        Instructions: []
+        Host: uid
+        Active: true
+        Dates: {
+	        Start: timestamp
+	        End: timestamp
+        }
+        Tags: [],
+        Category: [],
+        Online: boolean
+        Stats: {
+            Likes: []
+            Rsvp: []
+            people_who_showed_up:
+        people_who_stayed_till_the_end:
+        }
+        Location: {
+            Geo: {
+                Geopoint: geopoint,
+                Geohash: geohash
+        },
+        Address: {
+            fullAddress: string,
+            fullStreet: string,
+            streetNum: string,
+            Street: string,
+            City: string,
+            State: string?,
+            Country: String
+        }
+    }
+
+ */
+
+import { firestore } from "../admin/admin";
+
+interface event {
+    title: string;
+    description: string;
+    instructions: [];
+    host: number;
+    active: boolean;
+    tags: [];
+    category: [];
+    online: boolean;
+    stats: {"likes" : [], "rsvp" : [], "showedUp" : number, "stayed" : number};
+    dates: {};
+    location: {"geo" : [], "address": { "fullAddress" : string
+                                        "fullStreet" : string
+                                        "streetNum" : number
+                                        "street" : string
+                                        "city" : string
+                                        "state" : string
+                                        "zip" : number
+                                        "country" : string
+                                       }
+    }
+    
+}
+
+exports.createEvent = async (request: any, response: any) => {
+    const data = JSON.parse(request.body);
+    firestore
+        .collection("events")
+        .doc(data.docName)
+        .set(data)
+        .then(() => {
+            response.json(data);
+        })
+        .catch((error: any) => {
+            response.send("Error creating event")
+    });
+};
+
+exports.getEvent = async (request: any, response: any) => {
+    await firestore
+      .collection("events")
+      .doc(request.params.event)
+      .get()
+      .then((data) => {
+        response.json(data);
+      })
+      .catch((error: any) => {
+        response.send("Event not found");
+    });
+};
+
+exports.deleteEvent = async (request: any, response: any) => {
+    await firestore
+      .collection("events")
+      .doc(request.params.events)
```

Comment:
The data will be sent in the body and not as query parameters usually
```suggestion
      .doc(request.body.events)
```

Also probably save the body initially so that you can apply the event interface that was created
```
const data: event = JSON.parse(request.body)
```